### DISCUSSION TITLE
Make config.onInit hooks like array of functions

### DISCUSSION THREAD

**Discussion by gubarez** [payloadcms/payload#1953] [Feature Requests & Ideas]
config.onInit is a function, but plugins may need to hook on this lifecycle event. Perhaps the best thing is to have lifecycle event system, but ATM it would be easier to implement onInit to be an array of functions.

**Comment by jmikrut** [comment]
Hey @gubarez — you can still use the pattern with plugins - you'll just have to _**call any incoming**_ `onInit` method first in your own custom `onInit`. Then you could perform whatever logic you need afterward. Does that make sense?

But I would certainly not be opposed to modifying that config property to accept an array as well!

**Reply by gubarez** [reply]
Thanks for the prompt reply!
Yeah, that's what I've done already, but it this is not a good pattern. I could live with that, but it would be better otherwise. Not sure what's the implicatin on the execution lifecycle tough. IMO it is better to think about a lifecycle event system in the long-term.
